bootstraptest/test_yjit.rb

```diff
@@ -1337,6 +1337,16 @@ def make_str(foo, bar)
   make_str("foo", 123)
 }
 
+# test that invalidation of String#to_s doesn't crash
+assert_equal 'meh', %q{
+  class String
+    def to_s
+      "meh"
+    end
+  end
+  "".to_s
```

Comment:
I don't think this test exercises the invalidation because the class definition will be a side-exit. I think we'd have to move this to a method and warm it before the redefinition.

Reply:
I'm pretty sure that when tested with --yjit-call-threshold=1 this will send the invalidation. Certainly that's what I saw locally on my AWS instance. I think you're right at the default call threshold, though.